mne/io/fieldtrip/tests/test_fieldtrip.py

```diff
@@ -0,0 +1,206 @@
+
+
+@testing.requires_testing_data
+@pytest.mark.filterwarnings('ignore::RuntimeWarning')
+@pytest.mark.parametrize('cur_system, version, use_info',
+                         all_test_params_epochs)
+def test_epoched(cur_system, version, use_info):
+    """Test comparing reading an Epochs object and the FieldTrip version."""
+    has_pandas = _check_pandas_installed(strict=False) is not False
+    test_data_folder_ft = get_data_paths(cur_system)
+    mne_epoched = get_epoched_data(cur_system)
+    if use_info:
+        info = get_raw_info(cur_system)
+        pytestwarning = {'expected_warning': None}
+    else:
+        info = None
+        pytestwarning = no_info_warning
+
+    cur_fname = os.path.join(test_data_folder_ft,
+                             'epoched_%s.mat' % (version,))
+    if has_pandas:
+        pandas = _check_pandas_installed()
+        if version == 'v73' and not _has_h5py():
+            with pytest.raises(ImportError):
+                mne.io.read_epochs_fieldtrip(cur_fname, info)
+            return
+        with pytest.warns(**pytestwarning):
+            epoched_ft = mne.io.read_epochs_fieldtrip(cur_fname, info)
+        assert isinstance(epoched_ft.metadata, pandas.DataFrame)
+    else:
+        with pytest.warns(None) as warn_record:
+            if version == 'v73' and not _has_h5py():
+                with pytest.raises(ImportError):
+                    mne.io.read_epochs_fieldtrip(cur_fname, info)
+                return
+            epoched_ft = mne.io.read_epochs_fieldtrip(cur_fname, info)
+            assert epoched_ft.metadata is None
+            assert_warning_in_record(pandas_not_found_warning_msg, warn_record)
+            if pytestwarning['expected_warning'] is not None:
+                assert_warning_in_record(pytestwarning['message'], warn_record)
+
+    mne_data = mne_epoched.get_data()[:, :, :-1]
+    ft_data = epoched_ft.get_data()
+
+    check_data(mne_data, ft_data, cur_system)
+    check_info_fields(mne_epoched, epoched_ft, use_info)
+
+
+@testing.requires_testing_data
+@pytest.mark.filterwarnings('ignore::RuntimeWarning')
+@pytest.mark.parametrize('cur_system, version, use_info', all_test_params_raw)
+def test_raw(cur_system, version, use_info):
+    """Test comparing reading a raw fiff file and the FieldTrip version."""
+    # Load the raw fiff file with mne
+    test_data_folder_ft = get_data_paths(cur_system)
+    raw_fiff_mne = get_raw_data(cur_system, drop_extra_chs=True)
+    if use_info:
+        info = get_raw_info(cur_system)
+        pytestwarning = {'expected_warning': None}
+    else:
+        info = None
+        pytestwarning = no_info_warning
+
+    cur_fname = os.path.join(test_data_folder_ft,
+                             'raw_%s.mat' % (version,))
+
+    if version == 'v73' and not _has_h5py():
+        with pytest.raises(ImportError):
+            mne.io.read_raw_fieldtrip(cur_fname, info)
+        return
+    with pytest.warns(**pytestwarning):
```

Comment:
what are the warnings you want to catch?

Reply:
`pytestwarning` gets defined above (lines 125 to 130):

```python
    if use_info:
        info = get_raw_info(cur_system)
        pytestwarning = {'expected_warning': None}
    else:
        info = None
        pytestwarning = no_info_warning
```

if no info field has been supplied, `read_raw_fieldtrip` is supposed to throw a warning. this is checked for here.

Reply:
I don't see why `@pytest.mark.filterwarnings('ignore::RuntimeWarning')` and then catch it like this. 

I would use `recwarn` pytest feature and then inspect it here with all the logic needed. IMHO I think it would be cleaner.

Reply:
@massich the `@pytest.mark.filterwarnings('ignore::RuntimeWarning')` is included for all tests because reading raw data from all the various systems results in a broad variety of `RuntimeWarning`s, especially as the tests reuse raw data from the mne testing data, which is sometimes rather stripped down, leading to said warnings.